ui/src/features/Dashboard/components/MainSection.tsx

```diff
@@ -15,16 +16,21 @@ import {
   ContextMenuItem,
   ContextMenuTrigger,
 } from "@flow/components";
+import { useProject } from "@flow/lib/gql";
 import { useT } from "@flow/lib/i18n";
-import { generateProjects } from "@flow/mock_data/projectData";
-import { useCurrentProject, useCurrentWorkspace } from "@flow/stores";
+import { generateWorkflows } from "@flow/mock_data/workflowData";
+import { useCurrentProject, useCurrentWorkspace, useDialogType } from "@flow/stores";
 import type { Project } from "@flow/types";
+import { formatDate } from "@flow/utils";
 
 const MainSection: React.FC = () => {
   const t = useT();
   const [currentWorkspace] = useCurrentWorkspace();
   const [currentProject, setCurrentProject] = useCurrentProject();
   const navigate = useNavigate({ from: "/workspace/$workspaceId" });
+  const { useGetProjects, deleteProject } = useProject();
+  const { projects } = useGetProjects(currentWorkspace?.id as string);
```

Comment:
As discussed on Slack, do not typecast a type on a possible undefined.

Reply:
I passed it as a prop since it needs to exist before we can render this component.

Reply:
We are setting the state in the parent. I am not sure how you feel about using global state at too many places?

Reply:
SETTING global state should be only in very deliberate places so its easy to track. 
USING global state I think can, and should, be used liberally. That way we can avoid prop drilling and components are more mobile if we chose to rearrange the UI, etc